File: problems/10031/solution.cpp
// Accepted: 4ms (83.33%), 19.32MB (33.33%)

class Solution {
public:
    int missingInteger(vector<int>& nums) {
        unordered_set<int> numsSet(nums.begin(), nums.end());

        int seqEnd = 0;
        for (int i = 1; i < nums.size(); ++i) {
            if (nums[i] == nums[i-1] + 1) {
                seqEnd++;
            } else {
                break;
            }
        }

        int seqSum = 0;
        for (int i = 0; i <= seqEnd; ++i) {
            seqSum += nums[i];
        }

        while (true) {
            if (numsSet.count(seqSum) == 0) {
                return seqSum;
            }
            seqSum++;
        }

        return 0;
    }
};

File: problems/10031/solution.py
# Accepted: 33ms (100.00%), 17.33MB (40.00%)

class Solution:
    def missingInteger(self, nums: List[int]) -> int:
        numsSet = set(nums)

        # get the sum for the sequential prefix
        seqEnd = 0
        for i in range(1, len(nums)):
            if nums[i-1]+1 == nums[i]:
                seqEnd = i
            else:
                break

        seqSum = sum(nums[:seqEnd+1])

        # return the smallest missing integer greater than or equal to the sum
        while True:
            if seqSum not in numsSet:
                return seqSum

            seqSum += 1
